ipatests/test_integration/test_sssd.py

```diff
@@ -108,3 +114,57 @@ def test_auth_cache_enabled_when_configured(self, user):
             assert self.is_auth_cached(self.users[user])
             time.sleep(10)
             assert not self.is_auth_cached(self.users[user])
+
+    @contextmanager
+    def filter_user_setup(self, user, provider='ipa'):
+        tasks.kinit_admin(self.master)
+        master = self.master
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        if user['name'] == 'fakeuser':
+            provider = 'nss'
+        filter = {'filter_users': '{0}'.format(user['name'])}
+        try:
+            tasks.modify_sssd_conf(
+                master, master.domain.name, filter, provider)
+            yield
+        finally:
+            sssd_conf_backup.restore()
+            tasks.clear_sssd_cache(self.master)
+
+    def is_user_filtered(self, user):
```

Comment:
This function should be eliminated and its contents should be moved to test method, based on following:
 * it is used only in one place
 * it contains whole test scenario except setup
 * test method contains only invocation of setup and this function